TypeCobol/Compiler/Diagnostics/TypeCobolChecker.cs

```diff
@@ -816,38 +859,106 @@ public static void CheckStatement(Node node)
-                
+                    DiagnosticUtils.AddError(node,
+                        "[Set [pointer1, pointer2 ...] UP|DOWN BY n] only support pointers.");
+
                 // Check sender (increment)
                 int outputResult; // not used
                 if (!int.TryParse(statement.SendingVariable.ToString(), out outputResult))
-                {// Not an integer
-                    var variable = node.GetDataDefinitionForQualifiedName(new URI(statement.SendingVariable.ToString()));
+                {
+                    // Not an integer
+                    var variable =
+                        node.GetDataDefinitionForQualifiedName(new URI(statement.SendingVariable.ToString()));
                     if (variable == null || variable.DataType.Name != "Numeric")
-                    {// Not an Variable or a notNumeric variable
+                    {
+                        // Not an Variable or a notNumeric variable
                         if (statement.SendingVariable.ArithmeticExpression == null)
                             // Not an arithmetic expressions
-                            DiagnosticUtils.AddError(node, "Increment only support integer values, numeric variables and arithmetic expressions");
+                            DiagnosticUtils.AddError(node,
+                                "Increment only support integer values, numeric variables and arithmetic expressions");
                     }
                 }
             }
         }
     }
+
+
+    public class GlobalStorageSectionChecker
+    {
+        public static void OnNode([NotNull] Node node)
+        {
+            var globalStorageSection = node as GlobalStorageSection;
+            if (globalStorageSection == null) return;
+
+            //Check if GlobalStorageSection is declared in main program Rule - GLOBALSS_ONLY_IN_MAIN 
+            if (!globalStorageSection.GetProgramNode().IsMainProgram)
+                DiagnosticUtils.AddError(node,
+                    "GLOBAL-STORAGE SECTION is only authorized in the main program of this source file.");
+
+            //Check every GlobalStorageSection DataDefinition (children)
+            foreach (var child in globalStorageSection.Children)
+            {
+                CheckGlobalStorageChildren(child);
+            }
+        }
+
+        private static void CheckGlobalStorageChildren(Node node)
+        {
+            var dataDefinition = node as DataDefinition;
+            if (dataDefinition == null) return;
+
+            //Check variable LevelNumber Rule - GLOBALSS_LIMIT48 
+            var data = dataDefinition.CodeElement as DataDefinitionEntry;
+            if (data?.LevelNumber != null && data.LevelNumber.Value == 77)
+                DiagnosticUtils.AddError(node,
+                    "Data must be declared between level 01 to 49, or equals to 66 or 88.");
+            if (!(dataDefinition.Parent is DataDefinition) && data?.LevelNumber != null &&
```

Comment:
This check must be done by a Checker common to all DataDefinition. It can be removed.